core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/command/handlers/SingleTransferProcessCommandHandler.java

```diff
@@ -43,6 +43,7 @@ public void handle(SingleTransferProcessCommand command) {
             throw new EdcException(format("Could not find TransferProcess with ID [%s]", transferProcessId));
         } else {
             if (modify(transferProcess)) {
+                transferProcess.setModified();
```

Comment:
what if the subclass does not invoke any `transitionXYZ()` method?

Reply:
@paullatzelsperger This should be done through a specific method on the `TransferProcess` class.
I know that currently it's not always like this, but the business logic that involves enclosed entities/value objects should always pass through the transfer process, because it should be its own responsibility to keep everything consistent, not a caller responsibility. Otherwise we will lose control of it.

Reply:
one way of ensuring that enclosed objects are only modified "through" the TP would be to return deep-copies, e.g. `TransferProcess.getDataRequest()` would return a _copy_ of the datarequest.
Whenever a modification to the DataRequest is necessary, there'd have to be a dedicated method on the TP for it.

Reply:
Since we're channeling all modifications of our stateful entities through the `CommandQueue` anyways, that would be the obvious choke point to update the modified field. BTW the command handlers are the only two usages from outside a `MutableEntity`, so I think we're good there.